### PR TITLE
add git hooks

### LINKED ISSUE #8: Precommit hook
A precommit hook with [composer-git-hooks](https://github.com/BrainMaestro/composer-git-hooks) would be nice.

### PR DESCRIPTION
**Pull request by sualko**
fix #8

### CODE REVIEW

**Comment by sualko** [comment]
Any thoughts?

File: composer.json [hunk 1]
```diff
@@ -34,14 +34,28 @@
...
   "scripts": {
-    "test": "vendor/bin/phpunit"
+    "test": "vendor/bin/phpunit",
+    "post-install-cmd": "vendor/bin/cghooks add --ignore-lock",
+    "post-update-cmd": "vendor/bin/cghooks update"
+  },
+  "extra": {
+    "hooks": {
+      "pre-commit": [
+        "vendor/bin/php-cs-fixer fix --dry-run --allow-risky=yes"
```
Comment (SamuelWei) [inline composer.json:53]: Why is it --dry-run?

Comment (sualko) [reply composer.json:53]: Because I only want to test if there are errors and not change anything.

Comment (SamuelWei) [reply composer.json:53]: Ok, are there plans to apply the coding style automatically or is that the responsibility of the developer? What exactly is our coding style?

Comment (sualko) [reply composer.json:53]: To apply cs-fixer automatically, we would need to stash all changes before and amend the changes afterwards. There is way to many things which could go wrong in my opinion.

> What exactly is our coding style?

We basically use the same as upstream. See the `.php_cs` file.

Comment (SamuelWei) [reply composer.json:53]: Thanks

**Comment by sualko** [comment]
@SamuelWei ready to merge?

**Comment by SamuelWei** [comment]
> @SamuelWei ready to merge?

image

What about changing the hooks to fix on commit and check-only on push. This way the developer can save his non-cs-conformant code and safely run cs-fixer. (like in this example: https://github.com/BrainMaestro/composer-git-hooks)

Should we add 'composer install' to 'post-checkout' as well?

**Comment by sualko** [comment]
> What about changing the hooks to fix on commit and check-only on push. 

Feel free to change it. At the end it doesn't matter.

> Should we add 'composer install' to 'post-checkout' as well?

Great idea.

**Pull request merged**